### DISCUSSION TITLE
Problem with artisan::all() and resource/action

### DISCUSSION THREAD

**Discussion by olivier-nolf** [filamentphp/filament#12224] [Help]
### Package

Panel builder

### Package Version

v3.2.62

### How can we help you?

Hi,
I'm trying to create an action/wizard where I retrieve all commands from artisan.
The idea was to have this done in a CommandService with artisan:all(). 
Like this: $commands = collect(Artisan::all())->sortBy('name');
I want to have this dynamically where I can retrieve the options/arguments. So whenever a command has been changed/added/removed, it's reflected in the frontend.

When I call this commandService and dd this collection, I see the correct output. 
But as soon as I remove the dd and use it for example as a select in my action/resource,
(as a closure, as another function, .... it does not really matter how),
I get the following error: _Call to a member function prepare() on null_

protected function getHeaderActions(): array
    {
              
        $commands = (new CommandService())->get();
        dd($commands);
        
        return [
            Action::make('create')
                ->steps([
                    Step::make('Name')
                        ->description('Create a new schedule')
                        ->schema([
                            ...
                            ])
                     ];
    }

Does anyone has an idea/workaround (worst case I would have to create a config file and lose the dynamic part :s)

I am creating a new discussion as I've tried all the recommended solutions found on this topic
Thanks in advance!

**Comment by leandrocfe** [comment]
Do you mean, this?

```php
...
->schema([
    Select::make('options')
        ->options(function () {
            $keys = array_keys(Artisan::all());
            $options = array_combine($keys, $keys);
            return $options;
        }),
]),
```
